File: kattis/sun_and_moon.py
# https://open.kattis.com/problems/sunandmoon
# 1s
# 1.7 Easy


# 122 chars: Top 10 place 5
import sys;a,b,c,d=map(int,"".join(sys.stdin.readlines()).split());y=0
while 1:
 if(y+a)%b+(y+c)%d==0:print(y);break
 y+=1

"""
# A different approach of reading the numbers: 123 chars
def f():return list(map(int,input().split()))
a,b=f();c,d=f();y=0
while 1:
 if(y+a)%b==0 and(y+c)%d==0:print(y);break
 y+=1
"""

"""
# 128 chars: Top 10 place 6
import sys;a,b,c,d=map(int,"".join(sys.stdin.readlines()).split());y=0
while 1:
 if(y+a)%b==0 and(y+c)%d==0:print(y);break
 y+=1
"""

"""
# 140 chars. Top 10 place 8 (and removing my place 8 for the Prolog program)
import sys;a,b,c,d=[int(i)for i in"".join(sys.stdin.readlines()).split()];y=0
while 1:
 if(y+a)%b==0 and(y+c)%d==0:print(y);break
 y+=1
"""

File: kattis/sun_and_moon.pl
% https://open.kattis.com/problems/sunandmoon
% 1s
% 1.7 Easy

main :-
    readln([S1,S2,M1,M2],end_of_file),
    s(0,S1,S2,M1,M2).
main.
s(Y,S1,S2,M1,M2) :-
    ( ( (Y+S1) mod S2 =:= 0,(Y+M1) mod M2 =:= 0) ->
        writeln(Y)
    ;
        Y1 is Y+1,
        s(Y1,S1,S2,M1,M2)
    ).

/*
% Compressed: 156 chars: Top 10, place 8
main:-readln([S1,S2,M1,M2],end_of_file),s(0,S1,S2,M1,M2).
s(Y,S1,S2,M1,M2):-(((Y+S1)mod S2=:=0,(Y+M1)mod M2 =:= 0)->writeln(Y);Y1 is Y+1,s(Y1,S1,S2,M1,M2)).
*/

    